cpp/src/Ice/TcpEndpointI.cpp

```diff
@@ -163,6 +163,7 @@ IceInternal::TcpEndpointI::options() const
     // format of proxyToString() before changing this and related code.
     //
     ostringstream s;
+    s.imbue(locale::classic()); // Ensure we use the C locale for the number formatting.
```

Comment:
Here, I would not use imbue, but use to_string or to_chars to convert the integer into a string.